vm-virtio/src/console.rs

```diff
@@ -342,17 +342,13 @@ impl VirtioConsoleConfig {
 
 /// Virtio device for exposing console to the guest OS through virtio.
 pub struct Console {
-    kill_evt: Option<EventFd>,
     pause_evt: Option<EventFd>,
-    avail_features: u64,
-    acked_features: u64,
     config: Arc<Mutex<VirtioConsoleConfig>>,
```

Comment:
You need to change this so that it takes an `Arc<Mutex<Vec<u8>>>`. This way you won't duplicate. The only thing you need is to cast the `Vec<u8>` into a `VirtioConsoleConfig` anytime the code tries to use a `VirtioConsoleConfig`.